plugins/database/mysql/mysql.go

```diff
@@ -316,6 +316,89 @@ func (m *MySQL) RotateRootCredentials(ctx context.Context, statements []string)
 	return m.RawConfig, nil
 }
 
+// SetCredentials uses provided information to set the password a user in the
+// database. Unlike CreateUser, this method requires a username be provided and
+// uses the name given, instead of generating a name. This is used for setting
+// the password of static accounts, as well as rolling back passwords in the
+// database in the event an updated database fails to save in Vault's storage.
+func (m *MySQL) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	if len(statements.Rotation) == 0 {
+		return "", "", errors.New("empty rotation statements")
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	// Grab the lock
+	m.Lock()
+	defer m.Unlock()
+
+	// Get the connection
+	db, err := m.getConnection(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Start a transaction
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return "", "", err
+	}
+	defer func() {
+		_ = tx.Rollback()
+	}()
+	// Return the secret
+
+	stmts := statements.Rotation
+	// Execute each query
+	for _, stmt := range stmts {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
```

Comment:
I thought I saw a helper for this somewhere...

Reply:
I think this is pretty much the same query in `CreateUser` (with a slightly different query helper). Maybe we can pull this bit out into a helper if it doesn't exist.

Reply:
I can't seem to find it. Not a blocker.

Reply:
I refactored `CreateUser` and `SetCredentials` to use a common helper method in ef4137009 to execute their statements, as they were the same except for the map. I left the other method (`RevokeUser`, `RotateRoot..`) as they didn't use prepared statements and transactions